upb/msg.c

```diff
@@ -137,6 +137,18 @@ const upb_msg_ext *_upb_msg_getext(const upb_msg *msg,
   return NULL;
 }
 
+void _upb_msg_clearext(upb_msg *msg, const upb_msglayout_ext *ext_l) {
+  upb_msg_internal *in = upb_msg_getinternal(msg);
+  if (!in->internal) return;
+  const upb_msg_ext *base =
+      UPB_PTR_AT(in->internal, in->internal->ext_begin, void);
+  upb_msg_ext *ext = (void*)_upb_msg_getext(msg, ext_l);
```

Comment:
this seems weird to me as a void* cast...

Reply:
Done.

I sometimes use `(void*)` as a sort of C equivalent of `auto`, in the sense that `(void*)` can decay to any kind of pointer in C, and the declaration on the left shows you where it will end up. I consider it somewhat equivalent to:

```
  auto ext = static_cast<upb_msg_ext*>(_upb_msg_getext(msg, ext_l));
```

But I agree it could look weird to people not familiar with this convention.
```